cassandra/cluster.py

```diff
@@ -5133,6 +5133,7 @@ def next(self):
         if not self.response_future._continuous_paging_session:
             self.fetch_next_page()
             self._page_iter = iter(self._current_rows)
+            return self.next()
```

Comment:
@ultrabug this is identical to the next line after the if, how this fixes anything ?, do we have a test case to demonstrate this ?

Reply:
It's not doing the same. It runs the whole method again (recursively). `next(self)` not only calls `next(self._page_iter)` but also has all the logic in lines 5127-5135.